### PR TITLE
Move Sort() into sort.go.

### PR DESCRIPTION
**Pull request by fejta**
Correctly document and add unit test

/assign @michelle192837

### CODE REVIEW

File: util/gcs/sort.go [hunk 1]
```diff
@@ -93,3 +94,19 @@ func Touch(ctx context.Context, client ConditionalClient, path Path, generation
 	cond.DoesNotExist = true
 	return client.If(&cond, &cond).Upload(ctx, path, genZero, DefaultACL, "no-cache")
 }
+
+// Sort the builds by monotonically decreasing original prefix base name.
+//
+// In other words,
```
Comment (chases2) [inline util/gcs/sort.go:100]: Please add this as a test case also. The current coverage doesn't make it clear how, for example, b/10 and c/1 should interact.

**Review by chases2 — APPROVED** [review]
/hold for author review

File: util/gcs/sort.go (hunk 1 above)
Comment (fejta) [reply util/gcs/sort.go:100]: Please read the other tests in this package, which already cover this scenario.

Ensuring that the `Build` for `gs://c/10` sets the `baseName` to 10 is not the responsibility of `Sort()`

**Review by fejta — COMMENTED** [review]
/hold cancel

**Pull request merged**